src/ucp/wireup/address.c

```diff
@@ -137,21 +137,21 @@ ucp_address_gather_devices(ucp_worker_h worker, uint64_t tl_bitmap, int has_ep,
                                      devices, &num_devices);
 
         iface_attr = &worker->ifaces[i].attr;
-        if (iface_attr->cap.flags & UCT_IFACE_FLAG_CONNECT_TO_IFACE) {
-            dev->tl_addrs_size += iface_attr->iface_addr_len;
-        } else if (iface_attr->cap.flags & UCT_IFACE_FLAG_CONNECT_TO_EP) {
-            if (has_ep) {
-                dev->tl_addrs_size += iface_attr->ep_addr_len;
-            } else {
-                /* Empty address */
-            }
-        } else  {
+
+        if (!(iface_attr->cap.flags & UCT_IFACE_FLAG_CONNECT_TO_IFACE) &&
+            !(iface_attr->cap.flags & UCT_IFACE_FLAG_CONNECT_TO_EP)) {
             continue;
         }
 
+        dev->tl_addrs_size += iface_attr->iface_addr_len;
+
+        if (!(iface_attr->cap.flags & UCT_IFACE_FLAG_CONNECT_TO_IFACE) && has_ep) {
+            dev->tl_addrs_size += iface_attr->ep_addr_len;
+        }
+
         dev->tl_addrs_size += sizeof(uint16_t); /* tl name checksum */
         dev->tl_addrs_size += sizeof(ucp_address_packed_iface_attr_t); /* iface attr */
-        dev->tl_addrs_size += 1;                /* address length */
+        dev->tl_addrs_size += 2;                /* iface and ep address lengths */
```

Comment:
can add flag for tl_addr_len, in addition to UCP_ADDRESS_FLAG_LAST, which would indicate if ep address is present. this way no need to increase address size for modex